src/main/java/com/github/wovnio/wovnjava/HtmlConverter.java

```diff
@@ -168,9 +180,17 @@ private void appendHrefLang() {
     }
 
     private void replaceNodeToMarkerComment(Element element) {
-        Comment comment = new Comment(WOVN_MARKER_PREFIX + String.valueOf(markers.size()));
-        element.replaceWith(comment);
-        markers.add(restore(element.outerHtml())); // restore original text if element has marker
+        int commentId = this.markerManager.getCommentId();
+        Element newElement = restore(element);
+        if (this.markerManager.addComment(newElement, String.valueOf(commentId)) != commentId) {
+            throw new ConcurrentModificationException("Invlid Comment ID. CommentManger is not thread-safe.");
```

Comment:
I would not bother with this check. We are not creating any threads, so you may assume a request is processed consistently within a single thread.